src/steadystateproblem.cpp

```diff
@@ -433,41 +441,36 @@ realtype SteadystateProblem::getWrmsNorm(const AmiVector &x,
                        ewt.getNVector());
 }
 
-bool SteadystateProblem::checkConvergence(const Solver *solver,
-                                          Model *model,
-                                          SensitivityMethod checkSensitivities) {
-    if (checkSensitivities == SensitivityMethod::adjoint) {
+realtype SteadystateProblem::getWrms(Model *model,
+                                          SensitivityMethod sensi_method) {
+    realtype wrms = INFINITY;
+    if (sensi_method == SensitivityMethod::adjoint) {
         /* In the adjoint case, only xQB contributes to the gradient, the exact
            steadystate is less important, as xB = xQdot may even not converge
            to zero at all. So we need xQBdot, hence compute xQB first. */
         computeQBfromQ(model, xQ_, xQB_);
         computeQBfromQ(model, xB_, xQBdot_);
-        wrms_ = getWrmsNorm(xQB_, xQBdot_, solver->getAbsoluteToleranceQuadratures(),
-                           solver->getRelativeToleranceQuadratures(), ewtQB_);
+        wrms = getWrmsNorm(xQB_, xQBdot_, atol_quad_, rtol_quad_, ewtQB_);
     } else {
         /* If we're doing a forward simulation (with or without sensitivities:
            Get RHS and compute weighted error norm */
         model->fxdot(t_, x_, dx_, xdot_);
-        wrms_ = getWrmsNorm(x_, xdot_, solver->getAbsoluteToleranceSteadyState(),
-                           solver->getRelativeToleranceSteadyState(), ewt_);
+        wrms = getWrmsNorm(x_, xdot_, atol_, rtol_, ewt_);
     }
-    bool converged = wrms_ < RCONST(1.0);
-
-    if (checkSensitivities != SensitivityMethod::forward)
-        return converged;
+    return wrms;
+}
 
+    
+realtype SteadystateProblem::getWrmsFSA(Model *model) {
     /* Forward sensitivities: Compute weighted error norm for their RHS */
+    realtype wrms = INFINITY;
     for (int ip = 0; ip < model->nplist(); ++ip) {
-        if (converged) {
-            sx_ = solver->getStateSensitivity(t_);
-            model->fsxdot(t_, x_, dx_, ip, sx_[ip], dx_, xdot_);
-            wrms_ = getWrmsNorm(
-                sx_[ip], xdot_, solver->getAbsoluteToleranceSteadyStateSensi(),
-                solver->getRelativeToleranceSteadyStateSensi(), ewt_);
-            converged = wrms_ < RCONST(1.0);
-        }
+        if (wrms >= conv_thresh)
```

Comment:
wouldn't it break at the first step because INFINITY >= 1?

Reply:
yes, good point!